src/openforms/submissions/models/submission.py

```diff
@@ -642,3 +648,31 @@ def get_prefilled_data(self):
                 variable.key: variable.value for variable in prefill_vars
             }
         return self._prefilled_data
+
+    def get_cosigner_email(self) -> str | None:
+        for form_step in self.form.formstep_set.select_related("form_definition"):
+            for (
+                component,
+                upload_info,
+                data_path,
+                configuration_path,
+            ) in iterate_data_with_components(
+                form_step.form_definition.configuration, self.data
+            ):
+                if component["type"] == "cosign":
+                    return glom(self.data, data_path, default=None)
```

Comment:
Whatever it becomes, we want to `@functools.lru_cache(maxsize=1)` it too; this function is called a few times directly or indirectly via `waiting_on_cosign`

Reply:
It's true, getting the data from the `SubmissionValueVariable` once we have the key could be a better approach!

So this could become:
- jsonpathquery to get the form definition with the cosign => from here we get the key of the cosign component
- retrieve the `SubmissionValueVariable` to get the value.

Reply:
Since we discussed that the json-path query shouldn't be used, for now I added the lru cache to the function. If this function causes performance problems, we can look into saving the key of the cosign component on the form or something.

Reply:
the json-path query is still a viable option, we just need to properly implement it as a [custom database function](https://dev.to/idrisrampurawala/writing-custom-django-database-functions-4dmb), which allows us to use the ORM and all the protections it has and avoid the `.raw` call.

That is also the proper way to support different database vendors in the future.